File: src/main/java/aurocosh/divinefavor/common/item/spell_bow/capability/ISpellBowHandler.java
package aurocosh.divinefavor.common.item.spell_bow.capability;

import net.minecraft.item.ItemStack;
import net.minecraftforge.items.ItemStackHandler;

import java.util.List;

public interface ISpellBowHandler {
    int getSelectedSlotIndex();
    void setSelectedSlotIndex(int index);

    void switchToNext();
    void switchToPrevious();

    ItemStack getSelectedStack();
    List<ItemStack> getPreviousStacks();
    List<ItemStack> getNextStacks();

    ItemStackHandler getStackHandler();

    int getState();
}

File: src/main/kotlin/aurocosh/divinefavor/common/item/talisman_tools/spell_bow/capability/ISpellBowHandler.kt
package aurocosh.divinefavor.common.item.talisman_tools.spell_bow.capability

import aurocosh.divinefavor.common.item.talisman_tools.ITalismanTool
import aurocosh.divinefavor.common.item.arrow_talismans.base.ItemArrowTalisman

interface ISpellBowHandler : ITalismanTool {
    fun getSelectedTalisman(): ItemArrowTalisman?
}
